tests/test_google_ads_bookmarks.py

```diff
@@ -137,71 +166,48 @@ def test_run(self):
                     self.assertIsNotNone(stream_bookmark_1)
                     bookmark_value_1 = stream_bookmark_1.get(expected_replication_key)
                     self.assertIsNotNone(bookmark_value_1)
                     self.assertIsInstance(bookmark_value_1, str)
-                    try:
-                        parsed_bookmark_value_1 = dt.strptime(bookmark_value_1, self.REPLICATION_KEY_FORMAT)
-                    except ValueError as err:
-                        raise AssertionError(
-                            f"Bookmarked value does not conform to expected format: {self.REPLICATION_KEY_FORMAT}"
-                        ) from err
+                    self.assertIsDateFormat(bookmark_value_1, self.REPLICATION_KEY_FORMAT)
 
                     # Verify bookmarks saved match formatting standards for sync 2
                     self.assertIsNotNone(stream_bookmark_2)
                     bookmark_value_2 = stream_bookmark_2.get(expected_replication_key)
                     self.assertIsNotNone(bookmark_value_2)
                     self.assertIsInstance(bookmark_value_2, str)
+                    self.assertIsDateFormat(bookmark_value_2, self.REPLICATION_KEY_FORMAT)
 
-                    try:
-                        parsed_bookmark_value_2 = dt.strptime(bookmark_value_2, self.REPLICATION_KEY_FORMAT)
-                    except ValueError as err:
-
-                        try:
-                            parsed_bookmark_value_2 = dt.strptime(bookmark_value_2, "%Y-%m-%dT%H:%M:%S.%fZ")
-                        except ValueError as err:
-
-                            raise AssertionError(
-                                f"Bookmarked value does not conform to expected formats: " +
-                                "\n Format 1: {}".format(self.REPLICATION_KEY_FORMAT) +
-                                "\n Format 2: %Y-%m-%dT%H:%M:%S.%fZ"
-                            ) from err
+                    # Verify the bookmark is set based on sync end date (today) for sync 1
+                    # (The tap replicaates from the start date through to today)
+                    parsed_bookmark_value_1 = dt.strptime(bookmark_value_1, self.REPLICATION_KEY_FORMAT)
+                    self.assertEqual(parsed_bookmark_value_1, today_datetime)
 
-                    # Verify the bookmark is set based on sync execution time
-                    self.assertGreaterEqual(parsed_bookmark_value_1, today) # TODO can we get more sepecifc with this?
-                    self.assertGreaterEqual(parsed_bookmark_value_2, today)
+                    # Verify the bookmark is set based on sync execution time for sync 2
+                    # (The tap replicaates from the manipulated state through to todayf)
+                    parsed_bookmark_value_2 = dt.strptime(bookmark_value_2, self.REPLICATION_KEY_FORMAT)
+                    self.assertEqual(parsed_bookmark_value_1, today_datetime)
```

Comment:
Copy pasta?  Shouldn't this test parsted_bookmark_value_2?

Reply:
Good catch, will fix